### ISSUE TITLE
Simulator showing a static image without animation

### ISSUE THREAD

**Issue by tyson-ni** [joonspk-research/generative_agents#52]
When I run the simulation and go to `http://localhost:8000/simulator_home`, the page just displays the static image below. The command `run 20` seems to run without obvious error but no characters and animation show up.

![download](https://github.com/joonspk-research/generative_agents/assets/13012669/942a862b-c05a-48fe-8d34-81f878511a5a)

**Comment by RuoyuG** [comment]
so do i. do you know how to fix it?

**Comment by edom18** [comment]
I have the same issue. I also checked whole city map using arrow keys but I wasn't able to see any animations. I also found out the list of characters didn't show information up.

**Comment by sayaphoenix** [comment]
同样的问题，而且不能保存

**Comment by www-Ye** [comment]
same issue, how to solve it

**Comment by EvoNexusX** [comment]
同样 的问题，prompt上面有输出，但是画面没有显示

**Comment by HaochenZhao** [comment]
Same issue. I thought reloading it in browser might update the image, but it just shows "Please start the backend first." instead of any images.

**Comment by HaochenZhao** [comment]
If you are developers from China mainland (like me), this problem may come from openai api failure. When excuting step 3 in README.md, frontend server keeps printing ""POST /update_environment/ HTTP/1.1" 200 14" and nothing new appears in backend server. At least it works this way on my machine. To solve it, I add these code in reverie/backend_server/utils.py:
`PROXY = 'socks5h://127.0.0.1:7890'  
os.environ['HTTP_PROXY'] = os.environ['http_proxy'] = PROXY  
os.environ['HTTPS_PROXY'] = os.environ['https_proxy'] = PROXY  
os.environ['NO_PROXY'] = os.environ['no_proxy'] = '127.0.0.1,localhost,.local'  `,
then it seems to be ok now.

**Comment by edom18** [comment]
I solved the issue by creating `movement` folder. You can refer how to solve this to https://github.com/joonspk-research/generative_agents/pull/55

The problem is `reverie.py` didn't check and create target directory.

**Comment by ferencdv** [comment]
Thanks, but where exactly should I put a "movement" folder in the directory structure? Should I put anything else in the folder or just make a folder. Apologies if this was covered somewhere else but I could not find it.

**Comment by ferencdv** [comment]
Thanks. I think it is discussed here: https://github.com/joonspk-research/generative_agents/issues/44#issuecomment-1676376850

**Comment by edom18** [comment]
Yes, you're right. You can fix that by adding codes to `reveire.py` to check or create the folder before accessing files under the folder. Please refer to #55.

**Comment by EthenZhang** [comment]
same issue , and i have tried to create movement folder following  https://github.com/joonspk-research/generative_agents/pull/55 , it doesn't work .

**Comment by tobenkyo123** [comment]
Just found out this simulator screen can be moved by the ←↑↓→ key, but still don't know how to move by cursor or zoom in and out.

**Comment by Flanua** [comment]
Same I added the code:
![5555](https://github.com/joonspk-research/generative_agents/assets/140022879/3cfa0b6d-9739-4b70-a5c1-52f900deceaa)
And still same issue:
![3333](https://github.com/joonspk-research/generative_agents/assets/140022879/d42f7647-ac77-40f1-b3b3-3e785b3ee0e6)
--------------------
![222](https://github.com/joonspk-research/generative_agents/assets/140022879/ac0bede6-0aeb-43bc-adf1-90c38d7cd3e2)

--------------------
![111](https://github.com/joonspk-research/generative_agents/assets/140022879/e365357f-2e4f-4e4b-8152-f6dfd3f0608b)
--------------------
![44444](https://github.com/joonspk-research/generative_agents/assets/140022879/447b1e69-a5d2-474f-aa4b-a8f1dca54a83)

------------------
I'm using GPT4ALL Generative Agents (Using local LLM)

**Comment by renzhangzhe** [comment]
same issue, how to solve it

**Comment by Flanua** [comment]
P.S: I also tried to create movement folder manually in case that's the problem (because that code that I inserted still doesn't creating that folder for some reason) and the new movement chunks (like 0,1,2,3,4,etc) in .json format in that folder are still not created automatically for my simulation and the simulation base: base_the_ville_isabella_maria_klaus so we can't see our agents and time moving we can only see it moving by playing saved by default replays because those replays like: July1_the_ville_isabella_maria_klaus-step-3-1 and etc have movement folder and those movement chunks in json format inside movement folder present.

**Comment by thulanxc** [comment]
they are sleeping in the first 2000 steps or more, be patient. After 6 am, IR will start to move.

**Comment by PiotrZak** [comment]
agents processing information while sleeping or having REM phase?

**Comment by Flanua** [comment]
> they are sleeping in the first 2000 steps or more, be patient. After 6 am, IR will start to move.

As you can see from my image it says that agent is working at the counter thought. But the agent didn't move at all since I started the simulation.
P.S: I also waited 30 min for agents to move but nothing happened.

**Comment by Flanua** [comment]
> agents processing information while sleeping or having REM phase?

They plan their scheduler and do everything they supposed to do but we can't see it actually happening we can only see their actions in the console window though.

**Comment by thulanxc** [comment]
> > agents processing information while sleeping or having REM phase?
> 
> They plan their scheduler and do everything they supposed to do but we can't see it actually happening we can only see their actions in the console window though.

Just wait for them. In the first step they will plan for a long time. When they are planning, they do not move. After they finish planning(it costs a long time), they will start to sleep, and in your cmd window you will see lines run fast. Then when it comes to the 2000th step or more, 6 am in the simulated town, IR will start to move. So just wait.

**Comment by thulanxc** [comment]
> > agents processing information while sleeping or having REM phase?
> 
> They plan their scheduler and do everything they supposed to do but we can't see it actually happening we can only see their actions in the console window though.

You can run a demo first to see what will happen if everything is ok

**Comment by thulanxc** [comment]
> agents processing information while sleeping or having REM phase?

I believe they do nothing while sleeping.

**Comment by Flanua** [comment]
> Just wait for them. In the first step they will plan for a long time. When they are planning, they do not move. After they finish planning(it costs a long time), they will start to sleep, and in your cmd window you will see lines run fast. Then when it comes to the 2000th step or more, 6 am in the simulated town, IR will start to move. So just wait.

They don't move at all I waited a whole hour till the simulation ended! Do you understand.
Here is the proof that agents do planned actions but they don't gets reflected visually except the console window:
![3333333333333](https://github.com/joonspk-research/generative_agents/assets/140022879/caa312de-ea4f-4f45-a747-77267754d485)

**Comment by Flanua** [comment]
> You can run a demo first to see what will happen if everything is ok

What demo? You mean replays? Default replays are working great because agents movement information being saved properly to the movement folder during the simulation. We are not talking about replay here we are talking about simulation not working as it should with local LLMs though. It may work with GPT4 but many countries have no access to it and it pointless to make it online service only and it should support local LLMs in the first place.

**Comment by sardanian** [comment]
I am having the same issue where the browser is not updating the gamestate to show anyone do any movements. It seems to be working as it is showing actions by the agents, but the map and characters and current time are not updating in any way.

**Comment by sardanian** [comment]
I am able to run the replays fine. But the simluation is not running when I am trying to see it live via: http://localhost:8000/simulator_home
I have to wait till the simulation finishes to be able to see replays. I'd like to be able to see the simulation as it is running.

**Comment by myrainbowandsky** [comment]
have you resolved?

> I am able to run the replays fine. But the simluation is not running when I am trying to see it live via: http://localhost:8000/simulator_home I have to wait till the simulation finishes to be able to see replays. I'd like to be able to see the simulation as it is running.

**Comment by myrainbowandsky** [comment]
> When I run the simulation and go to `http://localhost:8000/simulator_home`, the page just displays the static image below. The command `run 20` seems to run without obvious error but no characters and animation show up.
> 
> ![download](https://user-images.githubusercontent.com/13012669/260203711-942a862b-c05a-48fe-8d34-81f878511a5a.png)

have you resolved?

**Comment by juliuswang2024** [comment]
同样的问题依旧存在，画面不动

**Comment by zqgong** [comment]
“environment\frontend_server\templates\home\main_script.html“  change like this, which will show the whole image.

       // Phaser 3.0 global settings. 
	// Configuration meant to be passed to the main Phaser game instance. 
	// const config = {
	//   type: Phaser.AUTO,
	//   width: 1500,
	//   height: 800,
	//   parent: "game-container",
	//   pixelArt: true,
	//   physics: {
	//     default: "arcade",
	//     arcade: {
	//       gravity: { y: 0 } } },
	//   scene: {
	//     preload: preload,
	//     create: create,
	//     update: update } };

	const config = {
	  type: Phaser.AUTO,
	  width: 4480,
	  height: 3200,
	  parent: "game-container",
	  pixelArt: true,
	  physics: {
	    default: "arcade",
	    arcade: {
	      gravity: { y: 0 } } },
	  scene: {
	    preload: preload,
	    create: create,
	    update: update },
	  scale: {zoom: 0.5} };
